File: library/src/main/java/com/chad/library/adapter/base/entity/SectionEntity.java
package com.chad.library.adapter.base.entity;

import java.io.Serializable;

public abstract class SectionEntity<T> implements Serializable {

  public boolean isHeader;
  public T t;
  public String header;

  public SectionEntity(boolean isHeader, String header) {
    this.isHeader = isHeader;
    this.header = header;
    this.t = null;
  }

  public SectionEntity(T t) {
    this.isHeader = false;
    this.header = null;
    this.t = t;
  }
}

File: library/src/main/java/com/chad/library/adapter/base/entity/SectionEntity.kt
package com.chad.library.adapter.base.entity


/**
 * 带头部布局的实体类接口
 * 实体类请继承此接口；如果使用java，请使用[JSectionEntity]抽象类
 */
interface SectionEntity : MultiItemEntity {

    val isHeader: Boolean

    /**
     * 用于返回item类型，除了头布局外，默认只有[NORMAL_TYPE]一种布局
     * 如果需要实现 item 多布局，请重写此方法，返回自己的type
     */
    override val itemType: Int
        get() = if (isHeader) HEADER_TYPE else NORMAL_TYPE

    companion object {
        const val NORMAL_TYPE = -100
        const val HEADER_TYPE = -99
    }
}
